### PR TITLE
[APAM-18] rewrite Sample App

### PR DESCRIPTION
**Pull request by passyruan**

### CODE REVIEW

File: sample/src/main/java/com/airwallex/paymentacceptance/repo/LocalMockRepository.kt [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+                        "metadata" to mapOf(
+                            "id" to 1
+                        )
+                    )
+                )
+                val customerId = JSONObject(customerResponse.string())["id"].toString()
+                Settings.cachedCustomerId = customerId
+                customerId
+            } else {
+                Settings.cachedCustomerId!!
```
Comment (aw-hector) [inline sample/src/main/java/com/airwallex/paymentacceptance/repo/LocalMockRepository.kt:96]: This force unwrapping can be avoided by:
`
val cachedCustomerId = Settings.cachedCustomerId
if (cachedCustomerId.isNullOrEmpty()) {
} else {
     cachedCustomerId
}
`

File: sample/src/main/java/com/airwallex/paymentacceptance/ui/UIIntegrationActivity.kt [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+                getString(R.string.payment_successful),
+                getString(R.string.payment_successful_message)
+            )
+        }
+    }
+
+    companion object {
+        private const val TAG = "UIIntegrationActivity"
+
+        fun startActivity(context: Context) {
```
Comment (aw-hector) [inline sample/src/main/java/com/airwallex/paymentacceptance/ui/UIIntegrationActivity.kt:121]: Can this be written as a extension method for context like:
`fun <T: Any> Context.startActivity(clazz: KClass<T>) {
    startActivity(Intent(this, clazz::class.java))
}`

File: sample/src/main/java/com/airwallex/paymentacceptance/viewmodel/APIIntegrationViewModel.kt [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+    val airwallexPaymentStatus: LiveData<AirwallexPaymentStatus> = _airwallexPaymentStatus
+
+    private val _paymentMethodList = MutableLiveData<List<AvailablePaymentMethodType>>()
+    val paymentMethodList: LiveData<List<AvailablePaymentMethodType>> = _paymentMethodList
+
+    private val _paymentConsentList = MutableLiveData<List<PaymentConsent>>()
+    val paymentConsentList: LiveData<List<PaymentConsent>> = _paymentConsentList
+
+    override fun init(activity: Activity) {
+        airwallex = Airwallex(activity)
```
Comment (aw-hector) [inline sample/src/main/java/com/airwallex/paymentacceptance/viewmodel/APIIntegrationViewModel.kt:47]: Will this potentially cause circular reference? i.e. activity -> viewModel, viewModel -> airwallex -> activity

File: sample/src/main/java/com/airwallex/paymentacceptance/ui/base/BaseMvvmActivity.kt [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+import android.widget.ProgressBar
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.content.ContextCompat
+import androidx.lifecycle.ViewModelProvider
+import androidx.viewbinding.ViewBinding
+import com.airwallex.paymentacceptance.R
+import com.airwallex.paymentacceptance.viewmodel.base.BaseViewModel
+import java.lang.reflect.ParameterizedType
+
+abstract class BaseMvvmActivity<VB : ViewBinding, VM : BaseViewModel> : AppCompatActivity() {
```
Comment (aw-hector) [inline sample/src/main/java/com/airwallex/paymentacceptance/ui/base/BaseMvvmActivity.kt:18]: Mvvm is an architecture, maybe we can call it `BasePaymentActivity` instead?

File: sample/src/main/java/com/airwallex/paymentacceptance/Constant.kt [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+
+import com.airwallex.android.core.AirwallexCheckoutMode
+import com.airwallex.android.core.Environment
+import com.airwallex.android.core.model.Address
+import com.airwallex.android.core.model.PaymentConsent
+import com.airwallex.android.core.model.PaymentMethod
+import com.airwallex.android.core.model.PhysicalProduct
+import com.airwallex.android.core.model.Shipping
+
+val products = mutableListOf(
```
Comment (aw-hector) [inline sample/src/main/java/com/airwallex/paymentacceptance/Constant.kt:11]: Does it need to be mutable?

File: sample/src/main/java/com/airwallex/paymentacceptance/repo/LocalMockRepository.kt (hunk 1 above)
Comment (passyruan) [reply sample/src/main/java/com/airwallex/paymentacceptance/repo/LocalMockRepository.kt:96]: ok

File: sample/src/main/java/com/airwallex/paymentacceptance/ui/UIIntegrationActivity.kt (hunk 2 above)
Comment (passyruan) [reply sample/src/main/java/com/airwallex/paymentacceptance/ui/UIIntegrationActivity.kt:121]: OK

File: sample/src/main/java/com/airwallex/paymentacceptance/ui/base/BaseMvvmActivity.kt (hunk 4 above)
Comment (passyruan) [reply sample/src/main/java/com/airwallex/paymentacceptance/ui/base/BaseMvvmActivity.kt:18]: ok

**Review by aw-hector — APPROVED** [review]
Great work! 🚀

**Pull request merged**